From d257a1e8d58199702454e1c8ab39b8a20f0c090c Mon Sep 17 00:00:00 2001
From: Ewan Milne <emilne@redhat.com>
Date: Fri, 22 Aug 2014 17:12:34 -0400
Subject: [scsi] sd: fix a bug in deriving the FLUSH_TIMEOUT from the basic I/O timeout

Message-id: <1408727554-19429-1-git-send-email-emilne@redhat.com>
Patchwork-id: 88711
O-Subject: [RHEL7.1 PATCH e-stor] sd: fix a bug in deriving the FLUSH_TIMEOUT from the basic I/O timeout
Bugzilla: 1123906
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

upstream commit 26b9fd8b3452dcf0a8862e307ee23f442f63fb51

(Modified for RHEL7.1 -- minor edits due to patches applied out of order)

upstream commit 26b9fd8b3452dcf0a8862e307ee23f442f63fb51
Author: K. Y. Srinivasan <kys@microsoft.com>
Date:   Fri Jul 18 17:11:27 2014 +0200

    sd: fix a bug in deriving the FLUSH_TIMEOUT from the basic I/O timeout

    Commit ID: 7e660100d85af860e7ad763202fff717adcdaacd added code to derive the
    FLUSH_TIMEOUT from the basic I/O timeout. However, this patch did not use the
    basic I/O timeout of the device. Fix this bug.

    Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
    Reviewed-by: James Bottomley <JBottomley@Parallels.com>
    Signed-off-by: Christoph Hellwig <hch@lst.de>

Signed-off-by: Ewan D. Milne <emilne@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/sd.c b/drivers/scsi/sd.c
index dcb92a6..93b879d 100644
--- a/drivers/scsi/sd.c
+++ b/drivers/scsi/sd.c
@@ -846,7 +846,7 @@ static int sd_setup_write_same_cmnd(struct scsi_device *sdp, struct request *rq)
 
 static int scsi_setup_flush_cmnd(struct scsi_device *sdp, struct request *rq)
 {
- rq->timeout *= SD_FLUSH_TIMEOUT_MULTIPLIER;
+ rq->timeout = rq->q->rq_timeout * SD_FLUSH_TIMEOUT_MULTIPLIER;
  rq->retries = SD_MAX_RETRIES;
  rq->cmd[0] = SYNCHRONIZE_CACHE;
  rq->cmd_len = 10;
-- 
1.7.1